### ISSUE TITLE
extremely slow loading of Panasonic DC-G9 80 MP files

### ISSUE THREAD

**Issue by heckflosse** [Beep6581/RawTherapee#5336]
The loading of this [file](https://raw.pixls.us/getfile.php/2584/nice/Panasonic%20-%20DC-G9%20-%204:3.RW2) is extremely slow on my system using latest RT dev.

Can someone please confirm this using this patch:
```diff
diff --git a/rtexif/rtexif.cc b/rtexif/rtexif.cc
index 571fd4e6b..b9d01e921 100644
--- a/rtexif/rtexif.cc
+++ b/rtexif/rtexif.cc
@@ -32,6 +32,8 @@
 
 #include "rtexif.h"
 
+#define BENCHMARK
+#include "../rtengine/StopWatch.h"
 #include "../rtengine/procparams.h"
 
 #include "../rtgui/cacheimagedata.h"
@@ -2778,6 +2780,7 @@ void ExifManager::parseStd (bool skipIgnored) {
 
 void ExifManager::parse (bool isRaw, bool skipIgnored)
 {
+    BENCHFUN
     int ifdOffset = IFDOffset;
 
     if (!f) {

```

```
parse took 6898 ms
```

7 seconds to parse the exif data is really strange...

**Comment by Beep6581** [comment]
@heckflosse
`5.6-47-g`b2936d185
x86_64 Intel(R) Core(TM) m3-6Y30 CPU @ 0.90GHz
```
$ /home/morgan/programs/rawtherapee/rawtherapee ~/downloads/Panasonic\ -\ DC-G9\ -\ 4_3.RW2
parse took 4794 ms
parse took 4795 ms
parse took 4708 ms
parse took 4708 ms
parse took 0 ms
```

**Comment by Floessie** [comment]
@heckflosse @Beep6581 How does it behave with [`metadata-exiv2`](https://github.com/Beep6581/RawTherapee/tree/metadata-exiv2)?

I'll give it a try in a VM...

**Comment by Floessie** [comment]
### dev
```
parse took 3570 ms
parse took 3571 ms
parse took 2618 ms
parse took 2618 ms
parse took 46 ms
parse took 2581 ms
parse took 2581 ms
```

### metadata-exiv2
```
FramesData took 77 ms
FramesData took 4 ms
FramesData took 3 ms
FramesData took 8 ms
FramesData took 9 ms
```

Best,
Flössie

**Comment by heckflosse** [comment]
Fixed with ffa461d
